src/modules/mahjong/HandValidator/HandValidator.ts

```diff
@@ -0,0 +1,249 @@
+              melds.push({
+                tiles: meldTiles,
+                type: MeldTypes.CONSECUTIVE,
+              });
+            }
+          }
+        }
+      });
+
+      // Start trying to create all triplet melds
+      if (passed) {
+        Object.keys(copyRemainingTiles).forEach((key) => {
+          if (copyRemainingTiles[key] === this.TRIPLET_SIZE) {
+            const meld = {
+              tiles: [key, key, key],
+              type: MeldTypes.TRIPLET,
+            };
+            copyRemainingTiles[key] -= this.TRIPLET_SIZE;
+            melds.push(meld);
+          }
+        });
+
+        if (melds.length === this.REQUIRED_NUM_OF_MELDS) {
+          results.valid.push({
+            ...vp,
+            melds,
+          });
+        }
+      } else {
+        results.invalid.push(vp);
+      }
+    });
+
+    return results;
+  }
+
+  // /**
+  // TO BE DONE AT A LATER PR
+  //  * Given a hand, determine how many points the hand is able to get based on
+  //  * what valid hands are possible to create
+  //  * ex: All triplets, all consec, purity, semi-pure
+  //  * @param validPair
+  //  * @returns an number representing the point value
+  //  */
+  // public static determineHandPoints(validPair: ValidPair): number {
+  //   console.log(validPair);
+  //   return 37;
+  // }
+
+  // /**
+  //  * Given a hand, determine how many extra points the hand is able to get
+  //  * ex: dragons, winds, flowers
+  //  * @param validPair
+  //  */
+  // public static determineExtraPoints(validPair: ValidPair): number {
+  //   console.log(validPair);
+  //   return 37;
+  // }
+
+  // /**
+  //  * Given HandStructureResults, calculate the total number of points for each valid hand
+  //  * @param handStructureResults
+  //  */
+  // public static evaluatePoints(handStructureResults: HandStructureResults) {}
+
+  /**
+   * Validates the Length of the hand must be 14
+   * @param tiles An array of tile objects
+   * @returns boolean, true if hand length is 14, otherwise false
+   */
+  public static validateHandLength(tiles: Tile[]): boolean {
+    if (tiles.length !== 14) return false;
+    return true;
+  }
+
+  /**
+   * Validates if the hand is a valid thirteen orphans hand
+   * @param tiles An array of tile objects
+   * @returns true if the hand is valid, otherwise false
+   */
+  public static validateThirteenOrphans(tiles: Tile[]): boolean {
+    if (!this.validateHandLength(tiles)) {
+      return false;
+    }
+
+    const thirteenOrphansTileRequirement: { [index: string]: number } = {
+      '1_DOT': 0,
+      '9_DOT': 0,
+      '1_BAMBOO': 0,
+      '9_BAMBOO': 0,
+      '1_CHARACTER': 0,
+      '9_CHARACTER': 0,
+      EAST: 0,
+      SOUTH: 0,
+      WEST: 0,
+      NORTH: 0,
+      REDDRAGON: 0,
+      GREENDRAGON: 0,
+      WHITEDRAGON: 0,
+    };
+
+    let failed = false;
+
+    tiles.forEach((t) => {
+      if (Object.prototype.hasOwnProperty.call(thirteenOrphansTileRequirement, t.toString())) {
```

Comment:
can simplify to:
```js
if (thirteenOrphansTileRequirement.hasOwnProperty(t.toString())) {
  ...
}
```

Reply:
Yea it's weird. It doesn't work in typescript for some reason so have to use the prototype method.